### PR TITLE
refactor: remove old api code

### PR DESCRIPTION
**Pull request by Daanvdplas**
Only removes old code in the runtime so that it can no longer be used on testnet until API complete. Removing the code in other crates will be done on the `api` branch as soon as it is rebased on this PR because it is very different and feels like I would have to do the job twice.

### CODE REVIEW

File: runtime/devnet/src/extensions.rs [hunk 1]
```diff
@@ -49,7 +33,6 @@ where
 	fn call<E: Ext>(&mut self, env: Environment<E, InitState>) -> Result<RetVal, DispatchError>
```
Comment (peterwht) [inline runtime/devnet/src/extensions.rs:33]: clippy warning

File: runtime/devnet/src/extensions.rs [hunk 2]
```diff
@@ -213,11 +184,8 @@ where
 	let key: RuntimeStateKeys = env.read_as()?;
 
 	let result = match key {
-		RuntimeStateKeys::Nfts(key) => read_nfts_state::<T, E>(key, &mut env),
-		RuntimeStateKeys::ParachainSystem(key) => {
-			read_parachain_system_state::<T, E>(key, &mut env)
-		},
-	}?
+		_ => Vec::<u8>::default(),
```
Comment (peterwht) [inline runtime/devnet/src/extensions.rs:187]: clippy warning

File: runtime/testnet/src/extensions.rs [hunk 3]
```diff
@@ -44,7 +33,6 @@ where
 	fn call<E: Ext>(&mut self, env: Environment<E, InitState>) -> Result<RetVal, DispatchError>
```
Comment (peterwht) [inline runtime/testnet/src/extensions.rs:33]: clippy warning

File: runtime/testnet/src/extensions.rs [hunk 4]
```diff
@@ -202,11 +184,8 @@ where
 	let key: RuntimeStateKeys = env.read_as()?;
 
 	let result = match key {
-		RuntimeStateKeys::Nfts(key) => read_nfts_state::<T, E>(key, &mut env),
-		RuntimeStateKeys::ParachainSystem(key) => {
-			read_parachain_system_state::<T, E>(key, &mut env)
-		},
-	}?
+		_ => Vec::<u8>::default(),
+	}
```
Comment (peterwht) [inline runtime/testnet/src/extensions.rs:188]: clippy warning

**Review by peterwht — COMMENTED** [review]
looks good. Just a couple clippy warnings that should be addressed

**Comment by Daanvdplas** [comment]
As discussed elsewhere, we are ignoring those clippy warnings for now because they will be introduced back in with the pop api PR.

**Review by evilrobot-01 — APPROVED** [review]

**Review by peterwht — APPROVED** [review]

**Pull request merged**